devtools/server/actors/replay/connection.js

```diff
@@ -533,6 +542,47 @@ class Recording extends EventEmitter {
   }
 }
 
+let locationListener;
+
+function getLocationListener(browser) {
+  locationListener = {
+    onLocationChange(aWebProgress, aRequest, aLocation) {
+      if(!aWebProgress.isTopLevel) return;
+
+      // Always allow blank and new tab
+      if (aLocation.displaySpec === "about:blank" || aLocation.displaySpec === "https://app.replay.io/browser/new-tab") {
+        return;
+      }
+
+      fetch(getAPIServer(), {
```

Comment:
we should wrap all of this logic in a helper so that we can re-use it and it is cleaner

Reply:
Done!